File: 2024/Python/Day3/solutionA.py
import re

data = []

regex = r"mul\(\d+,\d+\)";
with open("../../data/day3.txt") as file:
    for line in file:
        for instance in re.findall(regex, line):
            [a,b] = instance[4:-1].split(",")
            data.append(int(a) * int(b))

print(sum(data))

File: 2024/JS/Day3/solutionA.js
const fs = require("fs");

fs.readFile("../../data/day3.txt", (err, data) => {
  const regex = /mul\(\d+,\d+\)/g;
  let input = data.toString();
  const result = input.match(regex).reduce((acc, item) => {
    const [a, b] = item
      .slice(4, -1)
      .split(",")
      .map((x) => parseInt(x));
    return acc + a * b;
  }, 0);
  console.log(result);
});
